### PR TITLE
App Logger

### LINKED ISSUE #282: Add Logger
Add convenience logger for easier app debugging.

### PR DESCRIPTION
**Pull request by Kharchevskyi**
close #282 

- [x] Add App Logger
- [x] Replace all prints and debugPrints
- [x] Replace legacy logs
- [x] Wrap default Swift print to prevent security issues

To print ALL messages
Configuration.default isAll = true

To print with log level
Configuration.default isAll = false
Configuration.default logLevel = (.level)

**Convenience usage**
`let logger = Logger.nested(in: Self.self, with: "Flow name")`
`logger.logDebug("check is user logged in")`

Will print message with some flow, for example "App Start", or "IMAP"
**⚙️[App Start][GlobalRouter][23:53:17] check is user logged in**
in this case it will be easy to filter message by some flow in debugger, or some errors which happened during some flow

**Usage**
`let logger = Logger.nested("App Start")`
`logger.logDebug("some message")`

**For static messages**
`Logger.logDebug("some message")`

**Nested logger**
_inside some class_
`let logger = Logger.nested(Self.self)`
`logger.logWarning("some")`
will print name of the class where print was called
"🔥[GlobalRouter][11:25:02] warning message"

**Example messages**
"🏷[App Start][11:25:02] verbose message"
"ℹ️[App Start][11:25:02] info message"
"⚙️[App Start][11:25:02] debug message"
"❗️App Start][11:25:02] error message"
"🔥[App Start][11:25:02] warning message"

Full message
"⚙️[/Users/antonkharchevskyi/Local/flowcrypt-ios/FlowCrypt/App/Logger.swift-logDebug(_:)-183][23:40:25] message to print

### CODE REVIEW

**Comment by Kharchevskyi** [comment]
**To print ALL messages**
Configuration.default isAll = true

**To print with log level**
Configuration.default isAll = false
Configuration.default logLevel = (.level)

**Usage**
`let logger = Logger.nested("App Start")`
`logger.logDebug("some message")`
**OR**
`Logger.logDebug("some message")`

Example messages
"🏷[11:25:02][App Start][GlobalRouter-proceed-56] message"
"ℹ️[11:25:02][App Start][GlobalRouter-proceed-56] message"
"⚙️[11:25:02][App Start][GlobalRouter-proceed-56] message"
"❗️[11:25:02][App Start][GlobalRouter-proceed-56] message"
"🔥[11:25:02][App Start][GlobalRouter-proceed-56] message"

Full message
"⚙️[23:40:25][/Users/antonkharchevskyi/Local/flowcrypt-ios/FlowCrypt/App/Logger.swift-logDebug(_:)-183] message to print

**Comment by Kharchevskyi** [comment]
Screenshot 2021-04-23 at 23 59 24

**Review by tomholub — APPROVED** [review]
Looks good to me, but tests seem to fail?

File: FlowCrypt/Functionality/Mail Provider/Imap/Imap+session.swift [hunk 1]
```diff
@@ -76,9 +78,14 @@ extension Imap {
 
     func disconnect() {
         let start = DispatchTime.now()
```
Comment (Kharchevskyi) [inline FlowCrypt/Functionality/Mail Provider/Imap/Imap+session.swift:80]: Am I right that it was for purposes of measuring time between for example disconnect started and finished?

Comment (Kharchevskyi) [reply FlowCrypt/Functionality/Mail Provider/Imap/Imap+session.swift:80]: @tomholub

Comment (tomholub) [reply FlowCrypt/Functionality/Mail Provider/Imap/Imap+session.swift:80]: Correct, start of action to end of action

Comment (Kharchevskyi) [reply FlowCrypt/Functionality/Mail Provider/Imap/Imap+session.swift:80]: Let me came up with better solution for his purpose? CFAbsoluteTime is better for this purpose

Comment (tomholub) [reply FlowCrypt/Functionality/Mail Provider/Imap/Imap+session.swift:80]: Sure. The point is just to figure out latency of each action, it's up to you how you think it's best achieved.

**Comment by Kharchevskyi** [comment]
close #301

**Comment by tomholub** [comment]
Thank you - let me know once ready for final review.

**Comment by Kharchevskyi** [comment]
Screenshot 2021-05-05 at 11 18 57

ready 👍

**Comment by tomholub** [comment]
What exactly is 0.5024.... ? :) could it say `502ms` for example? Anything below 1ms is usually not meaningful to care about.

**Comment by Kharchevskyi** [comment]
will print 
"ℹ️[Js][Core][20:13:13] JsContext took 0.566 ms to start"
or 
"ℹ️[Js][Core][20:13:13] smth took 1.234 sec to start"

**Review by tomholub — APPROVED** [review]
Looks good, if tests pass

**Pull request merged**